File: p5.js/perlin_circles/NoiseLoop.js
class NoiseLoop{
  constructor(width, min, max, x, y, z){
    this.width = width;
    this.x = x;
    this.y = y;
    this.z = z;
    this.min = min;
    this.max = max;
  }

  valueAt(a, nz){
    let z = nz;
    if(!nz){
      z = this.z;
    }
    let nx = map(cos(a),-1,1,this.x,this.x+this.width);
    let ny = map(sin(a),-1,1,this.y,this.y+this.width);
    return map(noise(nx, ny, z),0,1,this.min,this.max);
  }

  addZ(amount){
    this.z += amount;
  }
}

File: Python/voronoi/noise_loop.py
import math
from noise import snoise3


class noise_loop:
    def __init__(self, rad, min, max, cx, cy):
        self.rad = rad
        self.min = min
        self.max = max
        self.cx = cx
        self.cy = cy

    def eval(self, t, z):
        x = ((self.rad * math.cos(t*math.pi*2)) + self.cx)/100
        y = ((self.rad * math.sin(t*math.pi*2)) + self.cy)/100

        res = snoise3(x, y, z)

        bound_len = self.max - self.min
        res = (res + 1) * bound_len/2 + self.min
        return res






